src/app/[locale]/viewer/_components/submodel-elements/generic-elements/entity-components/RelationShipDetailsDialog.tsx

```diff
@@ -34,20 +34,28 @@ export function RelationShipDetailsDialog(props: RelationShipDetailsModalProps)
     useEffect(() => {
         async function _fetchSubmodels() {
             try {
-                const submodelRefs = (await getSubmodelReferencesFromShell(base64AasId as string)) as Reference[];
+                const response = await getSubmodelReferencesFromShell(base64AasId as string);
+                let submodelRefs: Reference[];
```

Comment:
Again, the whole try-catch approach doesnt really make sense here. Also, variable naming is wrong, e.g., const submodelDescriptors = await getSubmodelDescriptorsById(reference.keys[0].value), is wrong, it should be const response = await getSubmodelDescriptorsById(reference.keys[0].value), and then response.result is the submodelDescriptors.

Reply:
I added a new throw in order to restore functionality. Refactoring the code to use if-else is beyond scope due to missing tests and low relevance of the compare view